pkg/grpc/authn/interceptor.go

```diff
@@ -27,7 +27,7 @@ func (u contextUpdater) updateContext(ctx context.Context) (context.Context, err
 		if errors.Is(err, jwt.ErrExpired) {
 			log.Debug("Cannot extract identity: token expired")
 		} else {
-			logging.GetRateLimitedLogger().WarnL(ri.Hostname, "Cannot extract identity: %v", err)
+			logging.GetRateLimitedLogger().DebugL(ri.Hostname, "Cannot extract identity: %v", err)
```

Comment:
The check above should be doing the job for the noisy messages like
```
root logger: 2023/12/11 19:19:22.955125 logger.go:77: Warn: pkg/grpc/authn/tokenbased/extractor.go:40 - Token validation failed for hostname acs-cia0k15dhps1ojkmtme0.acs.rhcloud.com: square/go-jose/jwt: validation failed, token is expired (exp) - 5 log suppressed for limiter "acs-cia0k15dhps1ojkmtme0.acs.rhcloud.com"
root logger: 2023/12/11 19:19:22.955204 logger.go:77: Warn: pkg/grpc/authn/interceptor.go:30 - Cannot extract identity: token validation failed - 5 log suppressed for limiter "acs-cia0k15dhps1ojkmtme0.acs.rhcloud.com"
```

The reason it does not work any more is because we now redefine the error class in https://github.com/stackrox/stackrox/blob/2539abc34d7c897e3f87c0c2c314fb32ea851a1a/pkg/grpc/authn/tokenbased/extractor.go#L46

We should either fix the error class propagation (I don't remember why it was redefined) or get rid of this `if-else` and put a single log statement under `DebugL`. I'd prefer the former if it is reasonable.